app/src/main/java/org/mozilla/FirefoxLite/firefoxdownloader/MainActivity.java

```diff
@@ -0,0 +1,294 @@
+        //            startActivity(intent);
+        //        }
+        init();
+        onClickListenerOne();
+
+    }
+
+    private void requeststoragepermission() {
+        if (ActivityCompat.shouldShowRequestPermissionRationale(this, Manifest.permission.READ_EXTERNAL_STORAGE)) {
+            new AlertDialog.Builder(this).setTitle("Permission Needed").setPositiveButton("ok", new DialogInterface.OnClickListener() {
+                @Override
+                public void onClick(DialogInterface dialog, int which) {
+                    ActivityCompat.requestPermissions(MainActivity.this, new String[]{Manifest.permission.READ_EXTERNAL_STORAGE}, STORAGE_PERMISSIONS_CODE);
+                }
+            }).setNegativeButton("cancel", new DialogInterface.OnClickListener() {
+                @Override
+                public void onClick(DialogInterface dialog, int which) {
+                    dialog.dismiss();
+                }
+            }).create().show();
+        } else {
+            ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.READ_EXTERNAL_STORAGE}, STORAGE_PERMISSIONS_CODE);
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        if (STORAGE_PERMISSIONS_CODE == requestCode) {
+            if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                Toast.makeText(this, "PERMISSION GRANTED", Toast.LENGTH_SHORT).show();
+            } else {
+                Toast.makeText(this, "PERMISSION DENIES", Toast.LENGTH_SHORT).show();
+            }
+        }
+    }
+
+    private void init() {
+        buttonOne = findViewById(R.id.buttonOne);
+        buttonCancelOne = findViewById(R.id.buttonCancelOne);
+        buttonScheduleOne = findViewById(R.id.buttonscheduleOne);
+        textViewProgressOne = findViewById(R.id.textViewProgressOne);
+        progressBarOne = findViewById(R.id.progressBarOne);
+    }
+
+
+
+    public void onClickListenerOne() {
+        buttonOne.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick( View view) {
+                Intent serviceintent = new Intent(MainActivity.this, SampleService.class);
+                startService(serviceintent);
+            }
+        });
+
+        buttonCancelOne.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent serviceIntent = new Intent(MainActivity.this,SampleService.class);
+                serviceIntent.putExtra("downloadstatus", "cancelled");
+                startService(serviceIntent);
+            }
+        });
+        buttonScheduleOne.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+
+                if (schedulesession!=null && schedulesession.equals("true")) {
```

Comment:
this is the info related to Download Task, and it's not a UI state. So you shouldn't store it in the Activity. The activity also belongs to the UI layer.